Code/GraphMol/MolAlign/AlignMetrics.cpp

```diff
@@ -0,0 +1,131 @@
+// $Id$
+//
+//  Copyright (C) 2001-2008 Greg Landrum and Rational Discovery LLC
+//
+//   @@ All Rights Reserved @@
+//  This file is part of the RDKit.
+//  The contents are covered by the terms of the BSD license
+//  which is included in the file license.txt, found at the root
+//  of the RDKit source tree.
+//
+#include "AlignMolecules.h"
+#include <math.h>
+#include <Geometry/Transform3D.h>
+#include <Numerics/Vector.h>
+#include <GraphMol/Substruct/SubstructMatch.h>
+#include <GraphMol/Conformer.h>
+#include <GraphMol/ROMol.h>
+#include <Numerics/Alignment/AlignPoints.h>
+#include <GraphMol/MolTransforms/MolTransforms.h>
+
+namespace RDKit {
+namespace MolAlign {
+
+double getConformerRMS(ROMol &mol, unsigned int confId1, unsigned int confId2,
+                       const std::vector<unsigned int> *atomIds,
+                       bool prealigned) {
+  if (!prealigned) {
+    static const unsigned int confIds[] = {confId1, confId2};
+    const std::vector<unsigned int> confIds2(
+        confIds, confIds + sizeof(confIds) / sizeof(confIds[0]));
+    MolAlign::alignMolConformers(mol, atomIds, &confIds2);
+  }
+
+  const Conformer conf1 = mol.getConformer(confId1);
+  const Conformer conf2 = mol.getConformer(confId2);
+
+  const size_t nAtoms = conf1.getNumAtoms();
+  double ssr = 0.0;
+  for (unsigned int i = 0; i < nAtoms; ++i) {
+    RDGeom::Point3D v = conf1.getAtomPos(i) - conf2.getAtomPos(i);
+    ssr += v.dotProduct(v);
+  }
+  ssr /= nAtoms;
+  return sqrt(ssr);
+}
+
+double getBestRMS(const ROMol &refMol, ROMol &prbMol, int refCid, int prbCid,
+                  std::vector<MatchVectType> *atomMaps) {
+  RDGeom::Point3DConstPtrVect refPoints, prbPoints;
+
+  if (!atomMaps || atomMaps->size() == 0) {
```

Comment:
I have the second condition here, because of the way I solved the Python integration where I create an empty `std::vector<MatchVectType>` as default for missing atomMaps.